### PR TITLE
feat: Limit heap memory. Fix git user problem

### PR DESCRIPTION
**Pull request by fmarek-kindred**

### CODE REVIEW

File: lock-manager/deployment/helm/templates/deployment.yaml [hunk 1]
```diff
@@ -27,10 +27,10 @@ spec:
           resources:
             limits:
               cpu: "1"
-              memory: "128Mi"
+              memory: "256Mi"
```
Comment (fmarek-kindred) [inline lock-manager/deployment/helm/templates/deployment.yaml:30]: Started getting OOM on local, so bumped it up a little

File: k8s-deployer/scripts/publish-report.sh [hunk 2]
```diff
@@ -34,6 +32,13 @@ then
   exit 1
 fi
 
+if [ "${USERNAME}" == "" ];
```
Comment (gk-kindred) [inline k8s-deployer/scripts/publish-report.sh:35]: I don't think this variable has anything to do with git. `zsh` sets `$USERNAME` env variable. And when you are in bash, the variable doesn't have any value

image

File: k8s-deployer/scripts/publish-report.sh [hunk 3]
```diff
@@ -48,6 +53,9 @@ then
   exit 1
 fi
 
+# The caller is expected to set "USERNAME" env variable
+AUTHOR_ARG="${USERNAME} <${USERNAME}@kindredgroup.com>"
```
Comment (gk-kindred) [inline k8s-deployer/scripts/publish-report.sh:57]: As this is OS repo, and the report can be pushed to any git repo, we shouldn't have `@kindredgroup.com` as suffix to the author.

File: k8s-deployer/scripts/publish-report.sh (hunk 2 above)
Comment (fmarek-kindred) [reply k8s-deployer/scripts/publish-report.sh:35]: It is also set by jenkins

File: k8s-deployer/scripts/publish-report.sh (hunk 3 above)
Comment (fmarek-kindred) [reply k8s-deployer/scripts/publish-report.sh:57]: Yes makes sense. I will add a separate ticket to fix it.

File: lock-manager/deployment/helm/templates/deployment.yaml (hunk 1 above)
Comment (gk-kindred) [reply lock-manager/deployment/helm/templates/deployment.yaml:30]: Although this is fine for now, if we get OOM again, we may have to profile and see why we get OOM. What surprises me though is why Lock manager would run into OOM for 128MB. Lets see how it goes with 256MB

File: k8s-deployer/scripts/publish-report.sh [hunk 4]
```diff
@@ -63,13 +71,17 @@ CONTENT_DIR_PATH="../${CONTENT_DIR}"
 pwd
 
 git clone $GIT_REPO . && \
+  cp .git/config .git/config_backup && \
+  git config user.email "${USERNAME}@kindredgroup.com" && \
```
Comment (gk-kindred) [inline k8s-deployer/scripts/publish-report.sh:75]: Same here for `@kindredgroup.com` email suffix

Comment (fmarek-kindred) [reply k8s-deployer/scripts/publish-report.sh:75]: Yes, created ticket already.

**Review by gk-kindred — APPROVED** [review]

**Pull request merged**